### DISCUSSION TITLE
Accessing the same cache from multiple components

### DISCUSSION THREAD

**Discussion by Jake-Prentice** [TanStack/query#1313] [Q&A] [answered]
I'm fetching quizzes from the backend, I have information like the title and the contents, configurations ect...I'm loading it in at once when a sidebar opens: 

```
//loading data
const {data: quizzes, isLoading} = useQuery("quizzes", fetchQuizzes, {
     onError: err => setError(err.message)
});
```

I then want to access the contents of the quiz within another component - it looks like this 

![image](https://user-images.githubusercontent.com/18198033/99915025-0d817980-2cf9-11eb-97c3-db81781cd220.png)

I was wondering about best practices, I could pass the data down the tree, or create a context but I felt that's the reason I used react-query - to stop managing loading data in context/state. What I did was just fetch the data from the cache by doing -

```
./contentsCard

const quizzes = queryCache.getQueryData("quizzes");
```

but I don't get any of the isLoading, status things. Is this an acceptable way to do it?

**Comment by TkDodo** [comment, marked as answer]
The recommended way would be to just call `useQuery` again, or abstract the useQuery away in a custom `useQuizzesQuery` hook and call that whenever you need the data. React query will give you the data from the cache and subscribe you to it, and also perform a background refetch when the component mounts.

**Reply by Jake-Prentice** [reply]
thanks, I don't know why I didn't think of that, really helpful! :)

**Reply by Jake-Prentice** [reply]
I'm guessing if useQuery already sees that the cache point "quizzes" exists, it won't try and fetch using the "fetchQuizzes" function and the data will just be that cached quizzes?

**Reply by TkDodo** [reply]
it will re-fetch when a new observer mounts and the data is considered stale (depending on `staleTime`).

Please read these two sections in the docs on that topic:
https://react-query.tanstack.com/docs/guides/important-defaults
https://react-query.tanstack.com/docs/guides/caching

(the explanation by example in the caching guide is really good)

**Reply by Jake-Prentice** [reply]
Ok thanks 🙏

**Reply by John-Paul-R** [reply]
Links are dead

Updated links (for v3):
- https://tanstack.com/query/v3/docs/framework/react/guides/caching
- https://tanstack.com/query/v3/docs/framework/react/guides/important-defaults

**Comment by cgvalayev** [comment]
What about a case when I need to see if mutation is loading from different components to avoid duplicate requests?

**Reply by TkDodo** [reply]
That's what `useIsMutating` is for
